webflow-hacks/cms-hacks/58/58.ts

```diff
@@ -0,0 +1,66 @@
+// inject the youtube api script
+const tag = document.createElement('script');
+tag.src = 'https://www.youtube.com/iframe_api';
+document.head.insertAdjacentElement('afterbegin', tag);
+
+/**
+ * default function called by the youtube api
+ */
+function onYouTubeIframeAPIReady() {
+  const IFRAME_SELECTOR = '[fs-hacks-element="iframe"]';
+  const iFrameContainer = document.querySelector(IFRAME_SELECTOR);
+  if (!iFrameContainer) return;
+
+  const iFrames = iFrameContainer.querySelectorAll('iframe');
+  iFrames.forEach((iFrame, index) => {
+    // set iframe properties
+    const { src } = iFrame;
+    if (!src) return;
+
+    const newSrc = new URL(src);
+    if (!newSrc.origin.includes('youtube')) return;
+
+    newSrc.searchParams.append('enablejsapi', '1');
+
+    iFrame.src = newSrc.toString();
+    iFrame.id = 'dynamic' + index;
```

Comment:
Perhaps updating the `iFrame` id is redundant now that you're passing the DOM element instead of the ID to the `createPlayer` method